packages/utils/src/zod-errors.ts

```diff
@@ -0,0 +1,100 @@
+import type { Event, EventHint } from '@sentry/types';
+
+import { isError } from './is';
+import { truncate } from './string';
+
+// Simplified ZodIssue type definition
+interface ZodIssue {
+  path: (string | number)[];
+  message?: string;
+  expected?: string | number;
+  received?: string | number;
+  unionErrors?: unknown[];
+  keys?: unknown[];
+}
+
+interface FlattenedZodResult {
+  formErrors: unknown[];
+  fieldErrors: Record<string, unknown[]>;
+}
+
+interface ZodError extends Error {
+  issues: ZodIssue[];
+
+  get errors(): ZodError['issues'];
+  flatten(): FlattenedZodResult;
+}
+
+function originalExceptionIsZodError(originalException: unknown): originalException is ZodError {
```

Comment:
Actually, can we just move this into the `core/src/integrations/zoderrors.ts` file? No need to have this be in utils IMHO.